woocommerce/woocommerce-admin
# Decouple currency and numbers packages from `wcSettings`.

Fixes #3029.
Fixes #3032.

This PR seeks to remove the `wcSettings` dependency from the Currency and Number packages.

### `Currency`

The currency package has been rewritten to export a `Currency` class instead of several utility functions. It takes an object in the constructor for configuration - which matches the structure of the `wcSettings` currency options.

A new `lib/currency-format` module has been introduced that exports the same API as the previous version of `@woocommerce/currency` so that the consuming parts of the WC-Admin app don't need to change (too much).

### `Number`

The exported methods of the number package have been rewritten to accept a configuration object as their first parameter - this allows for composition using lodash's `partial()` - which is done in the new `lib/number-format` module.

A new `lib/number-format` module has been introduced that exports the same API as the previous version of `@woocommerce/number` so that the consuming parts of the WC-Admin app don't need to change (too much).

### `AdvancedFilters`

Make the `currency` prop required, and expect an instance of the new `Currency` object.

### Detailed test instructions:

- Verify that numbers and monetary values are properly formatted on all WC-Admin screens (check activity panels, reports, filters, etc)
- Verify that changing store currency settings changes the formatting
- Review changes to documentation



### Changelog Note:



Fix: decouple Currency and Number packages from global wcSettings object.
